src/object.d

```diff
@@ -1102,6 +1104,24 @@ class TypeInfo_Class : TypeInfo
         }
         return o;
     }
+
+   /**
+    * Returns true if the class described by `child` derives from the class
+    * described by this `TypeInfo_Class`.
+    *
+    * Params:
+    *  child = TypeInfo for some class
+    * Returns:
+    *  true if the class described by `child` derives from the class described
+    *  by `child`, false otherwise.
+    */
+    bool isBaseOf(scope const TypeInfo_Class child) const @nogc nothrow pure @trusted
+    {
+        for (auto ti = cast() child; ti !is null; ti = ti.base)
+            if (ti is this)
+                return true;
+        return false;
```

Comment:
If there's some case for classes where this gives the wrong answer but `_d_isbaseof` gives the right answer I haven't found it. Unless such a case can be found I'll leave it here since `_d_isbaseof` potentially does a lot more work and `extern(C)` function calls are opaque to the optimizer.

Reply:
Looking back at this, checking straight up the inheritance chain rather than using _d_isbaseof means that the receiver TypeInfo_Class and argument TypeInfo_Class must represent actual classes rather than being the .info field of some TypeInfo_Interface.

Reply:
Solution: detect an interface's ClassInfo (TypeInfo_Class that does not represent a class) by checking if the initializer has length zero.